Updated error handling logic

nithinsubhash opened the pull request:
- Added a flag to LookupError to mark it as fatal
- Added a field `message` to the Lookup struct. The error string is added to this field if the error is not fatal

nithinsubhash committed:
Updated error handling logic

- Added a flag to LookupError to mark it as fatal
- Added a field `message` to the Lookup struct. The error string is added to this field if the error is not fatal

lguegan commented on api/lookup.go:
```diff
@@ -16,10 +15,11 @@ type Lookup struct {
 	MD5Hash     string   `json:"md5Hash" xml:"md5Hash"`
 	ValidFormat bool     `json:"validFormat" xml:"validFormat"`
-	Deliverable string    `json:"deliverable" xml:"deliverable"`
+	Deliverable bool    `json:"deliverable" xml:"deliverable"`
```
The reason I had create a string here is there are some cases where we are not sure, such as ucla domains. The message might cover these cases though

McManCSU replied:
Do you want to keys here - boolean is simpler to test. Expanded message is also helpful?

nithinsubhash replied:
@lguegan I switched to boolean just for consistency and the ease of testing. But yes, I understand that a `true`/`false` could be misleading when trumail is unable to verify. Do you suggest that we stick to using strings?

@McManCSU Are you suggesting that we return just a boolean and a message instead of multiple keys?

nithinsubhash committed:
Updated echo version, added prometheus exporter

nithinsubhash committed:
Reduced MX dial timeout

Reduced MX server dial timeout to 50s to prevent ALB idle timeout of 60s from happening.

nithinsubhash committed:
Reduced timeout to 40s

nithinsubhash committed:
Increased timeout to 45s

nithinsubhash committed:
Added disposable emails txt

nithinsubhash committed:
Added disposable flag, reformatted code

nithinsubhash committed:
Use tag instead of branchname

nithinsubhash committed:
Removed some disposable email domains

nithinsubhash committed:
Version update

nithinsubhash committed:
Removed an ignored error

nithinsubhash committed:
Added error details in the response

nithinsubhash committed:
Fix to handle SPF errors separately
